### PR TITLE
Add support JUSTID flag to XCLAIM command

### PR DESCRIPTION
**Pull request by dengliming**
Resolves [https://github.com/redis/jedis/issues/2146](https://github.com/redis/jedis/issues/2146)

### CODE REVIEW

File: src/main/java/redis/clients/jedis/PipelineBase.java [hunk 1]
```diff
@@ -2221,8 +2222,22 @@
...
+  @Override
+  public Response<List<byte[]>> xclaimIds(byte[] key, byte[] group, byte[] consumername,
+      long minIdleTime, XClaimParams params, byte[]... ids) {
+    getClient(key).xclaimIds(key, group, consumername, minIdleTime, params, ids);
+    return getResponse(BuilderFactory.BYTE_ARRAY_LIST);
+  }
+
   public Response<Object> sendCommand(final String sampleKey, final ProtocolCommand cmd,
-      final String... args) {
+                                      final String... args) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/PipelineBase.java:2240]:
```suggestion
      final String... args) {
```

File: src/main/java/redis/clients/jedis/commands/JedisCommands.java [hunk 2]
```diff
@@ -559,6 +560,14 @@ default String setex(String key, int seconds, String value) {
   List<StreamEntry> xclaim( String key, String group, String consumername, long minIdleTime,
       long newIdleTime, int retries, boolean force, StreamEntryID... ids);
 
+  /**
+   *  XCLAIM <key> <group> <consumer> <min-idle-time> <ID-1> <ID-2>
+   *        [IDLE <milliseconds>] [TIME <mstime>] [RETRYCOUNT <count>]
+   *        [FORCE] JUSTID
+   */
+  List<StreamEntryID> xclaimIds(String key, String group, String consumername, long minIdleTime,
+      XClaimParams params, StreamEntryID... ids);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/JedisCommands.java:569]:
```suggestion
  /**
   *  XCLAIM <key> <group> <consumer> <min-idle-time> <ID-1> <ID-2> ... <ID-N>
   *        [IDLE <milliseconds>] [TIME <mstime>] [RETRYCOUNT <count>]
   *        [FORCE]
   */
  List<StreamEntry> xclaim(String key, String group, String consumername, long minIdleTime,
      XClaimParams params, StreamEntryID... ids);

    /**
   *  XCLAIM <key> <group> <consumer> <min-idle-time> <ID-1> <ID-2> ... <ID-N>
   *        [IDLE <milliseconds>] [TIME <mstime>] [RETRYCOUNT <count>]
   *        [FORCE] JUSTID
   */
  List<StreamEntryID> xclaimJustid(String key, String group, String consumername, long minIdleTime,
      XClaimParams params, StreamEntryID... ids);
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]
1. Use full `JUSTID` in method name.
2. It is possible to have a version without JUSTID but with XClaimParams.

**Comment by dengliming** [comment]
> It is possible to have a version without JUSTID but with XClaimParams.

@sazzad16 That make sense to me. Maybe I can do this in a separate pr?

**Comment by sazzad16** [comment]
> Maybe I can do this in a separate pr?

If you're more comfortable that way, then sure!

**Comment by sazzad16** [comment]
> Maybe I can do this in a separate pr?

@dengliming If you're more comfortable that way, then sure!

File: src/main/java/redis/clients/jedis/BinaryClient.java [hunk 3]
```diff
@@ -1713,6 +1714,37 @@ public void xclaim(byte[] key, byte[] groupname, byte[] consumername, long minId
...
+    System.arraycopy(bparams, 0, args, index, paramLength);
+    index += paramLength;
+    if (justId) {
+      args[index++] = Keyword.JUSTID.getRaw();
+    }
+    sendCommand(XCLAIM, args);
+  }
+
+  public void xclaim(byte[] key, byte[] groupname, byte[] consumername, long minIdleTime,
+      XClaimParams params, byte[][] ids) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BinaryClient.java:1739]:
```suggestion
      XClaimParams params, byte[]... ids) {
```

File: src/main/java/redis/clients/jedis/BinaryClient.java [hunk 4]
```diff
@@ -1713,6 +1714,37 @@ public void xclaim(byte[] key, byte[] groupname, byte[] consumername, long minId
...
+    xclaim(key, groupname, consumername, minIdleTime, params, ids, false);
+  }
+
+  public void xclaimJustId(byte[] key, byte[] groupname, byte[] consumername, long minIdleTime,
+      XClaimParams params, byte[][] ids) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BinaryClient.java:1744]:
```suggestion
      XClaimParams params, byte[]... ids) {
```

File: src/main/java/redis/clients/jedis/commands/JedisCommands.java [hunk 5]
```diff
@@ -559,6 +560,22 @@ default String setex(String key, int seconds, String value) {
   List<StreamEntry> xclaim( String key, String group, String consumername, long minIdleTime,
       long newIdleTime, int retries, boolean force, StreamEntryID... ids);
 
+  /**
+   *  XCLAIM <key> <group> <consumer> <min-idle-time> <ID-1> <ID-2>
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/JedisCommands.java:564]:
```suggestion
   *  XCLAIM <key> <group> <consumer> <min-idle-time> <ID-1> ... <ID-N>
```

File: src/main/java/redis/clients/jedis/commands/JedisCommands.java [hunk 6]
```diff
@@ -559,6 +560,22 @@ default String setex(String key, int seconds, String value) {
...
+   *        [IDLE <milliseconds>] [TIME <mstime>] [RETRYCOUNT <count>]
+   *        [FORCE]
+   */
+  List<StreamEntry> xclaim(String key, String group, String consumername, long minIdleTime,
+      XClaimParams params, StreamEntryID... ids);
+
+  /**
+   *  XCLAIM <key> <group> <consumer> <min-idle-time> <ID-1> <ID-2>
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/JedisCommands.java:572]:
```suggestion
   *  XCLAIM <key> <group> <consumer> <min-idle-time> <ID-1> ... <ID-N>
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

**Pull request merged**